project-fob/Controllers/AttendeeController.cs

```diff
@@ -26,21 +26,43 @@ public ActionResult Index()
 
         public void Fob(string value)
         {
-            HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
+            if (HaveCookieId())
+            {
+                string userId = GetCookieId();
+                HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
 
-            string meetingString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
+                string meetingString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
 
-            Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Include(x => x.Fobbed).ThenInclude(x => x.User).Single(f => f.Meeting.MeetingId == meetingString);
+                Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Single(f => f.Meeting.MeetingId == meetingString);
+                
+                fob.AddFob(userId);
+                db.SaveChanges();
 
-            HttpContext.Session.TryGetValue("sessionid", out var session);
+            }
+            else
+            {
+                Index();
```

Comment:
Why does this call the Index action here? I didn't expect that to do anything (because it just builds the view for the page, but the return value is ignored)